### PR TITLE
Add support for Equanimity

### PR DESCRIPTION
**Pull request by TyraelWasTaken**
## Description

Adds support for Equanimity keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tyraelwastaken/equanimity/config.h
Comment (zvecr) [inline keyboards/tyraelwastaken/equanimity/config.h:1]: This file should be removed.

File: keyboards/tyraelwastaken/equanimity/keyboard.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
+{
+    "manufacturer": "TyraelWasTaken",
+    "keyboard_name": "Equanimity",
+    "maintainer": "TyraelWasTaken",
+    "bootloader": "stm32-dfu",
+    "bootmagic": {
+        "enabled": true
+    },
```
Comment (zvecr) [inline keyboards/tyraelwastaken/equanimity/keyboard.json:8]:
```suggestion
```

File: keyboards/tyraelwastaken/equanimity/keyboard.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        "enabled": true,
+        "rotary": [
+            {"pin_a": "A9", "pin_b": "A10", "resolution": 1}
+        ]
+    },
+    "features": {
+        "audio": false,
+        "backlight": false,
+        "command": true,
+        "console": true,
```
Comment (zvecr) [inline keyboards/tyraelwastaken/equanimity/keyboard.json:20]:
```suggestion
        "bootmagic": true,
        "encoder": true,
        "mousekey": true,
```

File: keyboards/tyraelwastaken/equanimity/keyboard.json [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    "manufacturer": "TyraelWasTaken",
+    "keyboard_name": "Equanimity",
+    "maintainer": "TyraelWasTaken",
+    "bootloader": "stm32-dfu",
+    "bootmagic": {
+        "enabled": true
+    },
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "enabled": true,
```
Comment (zvecr) [inline keyboards/tyraelwastaken/equanimity/keyboard.json:11]:
```suggestion
```

File: keyboards/tyraelwastaken/equanimity/keyboard.json [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        ]
+    },
+    "features": {
+        "audio": false,
+        "backlight": false,
+        "command": true,
+        "console": true,
+        "extrakey": true,
+        "nkro": true,
+        "rgblight": false
```
Comment (zvecr) [inline keyboards/tyraelwastaken/equanimity/keyboard.json:23]:
```suggestion
        "nkro": true
```

File: keyboards/tyraelwastaken/equanimity/keyboard.json [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    },
+    "matrix_pins": {
+        "cols": ["B2", "B1", "B0", "A7", "A6", "A5", "A2", "B12", "B9", "B8", "B7", "B6", "B5", "B4", "B3", "A15"],
+        "rows": ["B11", "B10", "A1", "A3", "A4"]
+    },
+    "mouse_key": {
+        "enabled": true
+    },
```
Comment (zvecr) [inline keyboards/tyraelwastaken/equanimity/keyboard.json:31]:
```suggestion
```

File: keyboards/tyraelwastaken/equanimity/rules.mk
Comment (zvecr) [inline keyboards/tyraelwastaken/equanimity/rules.mk:1]: This file should be removed.

File: keyboards/tyraelwastaken/equanimity/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PGUP,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_HOME, KC_PGDN, KC_END
+    )  
+};
+
+#ifdef ENCODER_MAP_ENABLE
+    const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+        [_BL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+        [_FL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
```
Comment (zvecr) [inline keyboards/tyraelwastaken/equanimity/keymaps/default/keymap.c:33]:
```suggestion
        [_FL] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
```

File: keyboards/tyraelwastaken/equanimity/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PGUP,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_HOME, KC_PGDN, KC_END
+    )  
+};
+
+#ifdef ENCODER_MAP_ENABLE
+    const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+        [_BL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+        [_FL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
```
Comment (zvecr) [inline keyboards/tyraelwastaken/equanimity/keymaps/via/keymap.c:33]:
```suggestion
        [_FL] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/tyraelwastaken/equanimity/readme.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+* Hardware Supported: *UDB C3 or greater with Equanimity PCB, no other PCBs are supported for the case*
+* Hardware Availability: *If you wish to build this, all files are available here: https://github.com/TyraelWasTaken/Equanimity*
+
+Make example for this keyboard (after setting up your build environment):
+
+    make tyraelwastaken/equanimity:default
+
+Flashing example for this keyboard:
+
+    make TyraelWasTaken/equanimity:default:flash
```
Comment (fauxpark) [inline keyboards/tyraelwastaken/equanimity/readme.md:17]:
```suggestion
    make tyraelwastaken/equanimity:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/tyraelwastaken/equanimity/readme.md (hunk 8 above)
Comment (TyraelWasTaken) [reply keyboards/tyraelwastaken/equanimity/readme.md]: Updated: 903e082

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**